File: apps/web/src/catalog/types/dto/TemplateParamDTO.ts
/**
 * Miroir d'un paramètre de template dans `TemplateDetailDTO.params[]`.
 * Contrat figé : `{key, label, type, required, default_value, options, order_index}`.
 */
export interface TemplateParamDTO {
  key: string
  label: string
  /** Valeur brute de l'enum `param_type` (ex. « string », « select »). */
  type: string
  required: boolean
  /** Valeur par défaut sérialisée en texte, ou `null`. */
  default_value: string | null
  /** Choix possibles pour un paramètre de type `select`, sinon `null`. */
  options: string[] | null
  order_index: number
}

File: apps/api/app/catalog/presentation/schemas/template_param_dto.py
"""Schema DTO d'un parametre de template (vue detail)."""

from typing import Any

from pydantic import BaseModel, Field

from app.catalog.domain.enums.param_type import ParamType


class TemplateParamDTO(BaseModel):
    """Parametre de provisioning d'un template (exposee dans le detail)."""

    key: str = Field(..., description="Cle technique du parametre.")
    label: str = Field(..., description="Libelle affiche dans le formulaire.")
    type: ParamType = Field(..., description="Type pilotant le rendu du formulaire.")
    required: bool = Field(..., description="Parametre obligatoire.")
    default_value: str | None = Field(None, description="Valeur par defaut (texte), si definie.")
    options: dict[str, Any] | None = Field(
        None, description="Choix possibles pour un parametre de type SELECT."
    )
    order_index: int = Field(..., description="Ordre d'affichage dans le formulaire.")
